hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClient.java

```diff
@@ -135,10 +135,11 @@ boolean verifySignature(byte[] data, byte[] signature,
    *
    * @param pemEncodedCert        - pem encoded X509 Certificate
    * @param force                 - override any existing file
+   * @param caCert                - Is CA certificate.
    * @throws CertificateException - on Error.
    *
    */
-  void storeCertificate(String pemEncodedCert, boolean force)
+  void storeCertificate(String pemEncodedCert, boolean force, boolean caCert)
```

Comment:
Why don't you write a new function, called storeRootCertificate. That avoids adding this false argument to lots of other parts of the code.

Reply:
Agree, let's add a new function as @anuengineer suggested.